Panic in v2.22.2 - nil pointer dereference regarding integer conversion overflow checks.

Running `gosec -exclude-generated ./...` on my codebase resulted in:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10315e4bc]

goroutine 1 [running]:
go/token.(*File).Name(...)
	go/token/position.go:114
github.com/securego/gosec/v2/analyzers.newIssue({0x1036d9daa, 0x4}, {0x14006967170, 0x2a}, 0x140047e6b40, 0x0, 0x2, 0x1)
	github.com/securego/gosec/v2/analyzers/util.go:70 +0x19c
github.com/securego/gosec/v2/analyzers.runConversionOverflow(0x14000e5d6c0)
	github.com/securego/gosec/v2/analyzers/conversion_overflow.go:84 +0x328
github.com/securego/gosec/v2.(*Analyzer).CheckAnalyzers(0x140004dc580, 0x14003211a00)
	github.com/securego/gosec/v2/analyzer.go:447 +0x3b0
github.com/securego/gosec/v2.(*Analyzer).Process(0x140004dc580, {0x0, 0x0, 0x0}, {0x140001e61e0, 0x1d, 0x3d?})
	github.com/securego/gosec/v2/analyzer.go:318 +0x374
main.main()
	github.com/securego/gosec/v2/cmd/gosec/main.go:477 +0xbdc
make: *** [gosec] Error 2
```

I will try to provide some example code to trigger this panic shortly.

Edit: seems to be related to using `./...`. Using `<path>/...`, e.g. `/path/to/module/...` does not panic.

Appears to be easily triggered with code that fails to compile:

```go
package main

type Foo struct {
	SomeNumber int
}

type Bar struct {
	SomeNumber int32
}

func main() {
	foo := Foo{SomeNumber: 1}
	_ = Bar{SomeNumber: foo.SomeNumber}
}
```

```shell
❯ pwd
/Users/ccampo/code/ccampo/scratch/gosec_bug

❯ ls -lah
total 16
drwxr-xr-x@ 4 ccampo  staff   128B Mar 27 20:13 .
drwxr-xr-x@ 4 ccampo  staff   128B Mar 26 15:21 ..
-rw-r--r--@ 1 ccampo  staff    27B Mar 26 15:21 go.mod
-rw-r--r--@ 1 ccampo  staff   170B Mar 27 20:11 main.go

❯ cat go.mod
module gosecbug

go 1.24.1

❯ cat main.go
package main

type Foo struct {
	SomeNumber int
}

type Bar struct {
	SomeNumber int32
}

func main() {
	foo := Foo{SomeNumber: 1}
	_ = Bar{SomeNumber: foo.SomeNumber}
}

❯ gosec --version
Version: 2.22.2
Git tag:
Build date: 2025-03-05T10:40:35Z

❯ gosec -exclude-generated ./...
[gosec] 2025/03/27 20:14:26 Including rules: default
[gosec] 2025/03/27 20:14:26 Excluding rules: default
[gosec] 2025/03/27 20:14:26 Including analyzers: default
[gosec] 2025/03/27 20:14:26 Excluding analyzers: default
[gosec] 2025/03/27 20:14:26 Import directory: /Users/ccampo/code/ccampo/scratch/gosec_bug
[gosec] 2025/03/27 20:14:26 Checking package: main
[gosec] 2025/03/27 20:14:26 Checking file: /Users/ccampo/code/ccampo/scratch/gosec_bug/main.go
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10494a4bc]

goroutine 1 [running]:
go/token.(*File).Name(...)
	go/token/position.go:114
github.com/securego/gosec/v2/analyzers.newIssue({0x104ec5daa, 0x4}, {0x14000046720, 0x28}, 0x14000574c00, 0x0, 0x2, 0x1)
	github.com/securego/gosec/v2/analyzers/util.go:70 +0x19c
github.com/securego/gosec/v2/analyzers.runConversionOverflow(0x140002b20e0)
	github.com/securego/gosec/v2/analyzers/conversion_overflow.go:84 +0x328
github.com/securego/gosec/v2.(*Analyzer).CheckAnalyzers(0x14000b19000, 0x140004544e0)
	github.com/securego/gosec/v2/analyzer.go:447 +0x3b0
github.com/securego/gosec/v2.(*Analyzer).Process(0x14000b19000, {0x0, 0x0, 0x0}, {0x14000b1f460, 0x1, 0x3d?})
	github.com/securego/gosec/v2/analyzer.go:318 +0x374
main.main()
	github.com/securego/gosec/v2/cmd/gosec/main.go:477 +0xbdc
```

This code sample doesn't compile without an explicit conversion:

```
overflow-crash: > go build .
# github.com/securego/samples/overflow-crash
./main.go:13:22: cannot use foo.SomeNumber (variable of type int) as int32 value in struct literal
```

After applying the conversion, the gosec returns the expected warning without a crash:
```
package main

type Foo struct {
	SomeNumber int
}

type Bar struct {
	SomeNumber int32
}

func main() {
	foo := Foo{SomeNumber: 1}
	_ = Bar{SomeNumber: int32(foo.SomeNumber)}
}
```

```
overflow-crash: > gosec ./...
[gosec] 2025/04/08 09:32:22 Including rules: default
[gosec] 2025/04/08 09:32:22 Excluding rules: default
[gosec] 2025/04/08 09:32:22 Including analyzers: default
[gosec] 2025/04/08 09:32:22 Excluding analyzers: default
[gosec] 2025/04/08 09:32:22 Import directory: /usr/local/google/home/ccojocar/go/src/github.com/securego/samples/overflow-crash
[gosec] 2025/04/08 09:32:22 Checking package: main
[gosec] 2025/04/08 09:32:22 Checking file: /usr/local/google/home/ccojocar/go/src/github.com/securego/samples/overflow-crash/main.go
Results:


[/usr/local/google/home/ccojocar/go/src/github.com/securego/samples/overflow-crash/main.go:13] - G115 (CWE-190): integer overflow conversion int -> int32 (Confidence: MEDIUM, Severity: HIGH)
    12:         foo := Foo{SomeNumber: 1}
  > 13:         _ = Bar{SomeNumber: int32(foo.SomeNumber)}
    14: }

Autofix:

Summary:
  Gosec  : dev
  Files  : 1
  Lines  : 14
  Nosec  : 0
  Issues : 1
```

Anyhow, I will add a check for null file pointer to prevent this crash from happening.

--- a/analyzers/util.go
+++ b/analyzers/util.go
@@ -62,6 +62,7 @@ func newIssue(analyzerID string, desc string, fileSet *token.FileSet,
 	pos token.Pos, severity, confidence issue.Score,
 ) *issue.Issue {
 	file := fileSet.File(pos)
+	// This can occur when there is a compilation issue into the code.
 	if file == nil {
 		return &issue.Issue{}
 	}

--- a/analyzers/util.go
+++ b/analyzers/util.go
@@ -62,6 +62,9 @@ func newIssue(analyzerID string, desc string, fileSet *token.FileSet,
 	pos token.Pos, severity, confidence issue.Score,
 ) *issue.Issue {
 	file := fileSet.File(pos)
+	if file == nil {
+		return &issue.Issue{}
+	}
 	line := file.Line(pos)
 	col := file.Position(pos).Column
 

